lib/location/method_cloud_location.c

```diff
@@ -0,0 +1,441 @@
+	struct location_data location;
+	struct location_data location_result = { 0 };
+	struct method_cloud_location_start_work_args *work_data =
+		CONTAINER_OF(work, struct method_cloud_location_start_work_args, work_item);
+	const struct location_wifi_config *wifi_config = work_data->wifi_config;
+	const struct location_cellular_config *cell_config = work_data->cell_config;
+	int64_t scan_start_time;
+	int64_t scan_time;
+	int32_t used_timeout_ms;
+	int err = 0;
+
+	if (wifi_config && cell_config) {
+		used_timeout_ms = MIN(cell_config->timeout, wifi_config->timeout);
+	} else if (cell_config) {
+		used_timeout_ms = cell_config->timeout;
+	} else if (wifi_config) {
+		used_timeout_ms = wifi_config->timeout;
+	} else {
+		__ASSERT_NO_MSG(false);
+	}
+
+	location_core_timer_start(used_timeout_ms);
+	scan_start_time = k_uptime_get();
+
+	if (wifi_config) {
+		LOG_INF("Triggering WiFi Scanning");
+		err = method_wifi_scanning_start();
+		if (err) {
+			LOG_WRN("Cannot start Wi-Fi scanning, err %d", err);
+		}
+	} else {
+		k_sem_give(&wifi_scan_ready);
+	}
+
+	if (cell_config) {
+		LOG_INF("Triggering neighbor cell measurements");
+		err = method_cellular_ncellmeas_start();
+		if (err) {
+			LOG_WRN("Cannot start neighbor cell measurements");
+		}
+	} else {
+		k_sem_give(&ncellmeas_ready);
+	}
+
+	k_sem_take(&wifi_scan_ready, K_FOREVER);
+	k_sem_take(&ncellmeas_ready, K_FOREVER);
+	if (!running) {
+		goto end;
+	}
+
+	location_core_timer_stop();
+
+	/* Scannings done at this point of time. Store current time to response. */
+	location_utils_systime_to_location_datetime(&location_result.datetime);
+
+	/* Check if timeout is given */
+	params.timeout_ms = used_timeout_ms;
+	if (used_timeout_ms != SYS_FOREVER_MS) {
+		/* +1 to round the time up */
+		scan_time = (k_uptime_get() - scan_start_time) + 1;
+
+		/* Check if timeout has already elapsed */
+		if (scan_time >= used_timeout_ms) {
+			LOG_WRN("Timeout occurred during scannings");
+			err = -ETIMEDOUT;
+			goto end;
+		}
+		/* Take time used for neighbour cell measurements into account */
+		params.timeout_ms = used_timeout_ms - scan_time;
+	}
+
+	if (!location_utils_is_default_pdn_active()) {
+		/* Not worth to start trying to fetch the location over LTE.
+		 * Thus, fail faster in this case and save the trying "costs".
+		 */
+		LOG_WRN("Default PDN context is NOT active, cannot retrieve a location");
+		err = -EFAULT;
+		goto end;
+	}
+
+	params.service = cell_config ? cell_config->service : wifi_config->service;
+	if (cell_config) {
+		params.cell_data = &cell_data;
+	}
+
+	if (latest_scan_result_count > 1) {
+		/* Fill Wi-Fi scanning results: */
+		latest_wifi_info.cnt = latest_scan_result_count;
+		params.wifi_data = &latest_wifi_info;
+	} else {
+		params.wifi_data = NULL;
+		if (latest_scan_result_count == 1) {
+			/* Following statement seems to be true at least with HERE
+			 * (400: bad request).
+			 */
+			LOG_WRN("Retrieving a location based on a single Wi-Fi "
```

Comment:
There is a warning if only one wifi is found. For the sake of completeness, is there also a warning somewhere if 0 wifis or cells are found?

Reply:
`else` branch handles 0 WiFi APs, right?